modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/util/OpenAPIDeserializer.java

```diff
@@ -1490,7 +1490,8 @@ public Object getAnyExample(String nodeKey,ObjectNode node, String location, Par
                 }
             } if (example.getNodeType().equals(JsonNodeType.NUMBER)) {
                 Integer integerExample = getInteger(nodeKey, node, false, location, result);
-                if (integerExample != null) {
+                JsonNode format = node.get("format");
```

Comment:
Thanks!! @joeljons, I tried to push some changes but I have not permissions on your repo, so here is where I think it should be validated:

In the method getInteger()
like this (it would return null, and all integers will be validated as well):
line: 1196 : I added a boolean validation to make sure the number is an integer.

public Integer getInteger(String key, ObjectNode node, boolean required, String location, ParseResult result) {
        Integer value = null;
        JsonNode v = node.get(key);
        if (node == null || v == null) {
            if (required) {
                result.missing(location, key);
                result.invalid();
            }
        }
        else if(v.getNodeType().equals(JsonNodeType.NUMBER)) {
            if (v.isInt()) {
                value = v.intValue();
            }
        }
        else if(!v.isValueNode()) {
            result.invalidType(location, key, "integer", node);
        }
        return value;
    }